### PR TITLE
[Merged by Bors] - feat: port FieldTheory.SplittingField.IsSplittingField

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/FieldTheory/SplittingField/IsSplittingField.lean [hunk 1]
```diff
@@ -46,18 +46,27 @@ namespace Polynomial
...
+  adjoin_roots' : Algebra.adjoin K (↑(f.map (algebraMap K L)).roots.toFinset : Set L) = ⊤
 #align polynomial.is_splitting_field Polynomial.IsSplittingField
 
-variable {K L F}
-
 namespace IsSplittingField
 
+variable {K}
+
+-- Porting note: infer kinds are unsupported
```
Comment (jcommelin) [inline Mathlib/FieldTheory/SplittingField/IsSplittingField.lean:59]: What do you mean by these porting notes? Could you please expand them a bit?

Comment (Parcly-Taxel) [reply Mathlib/FieldTheory/SplittingField/IsSplittingField.lean:59]: I just lifted them from #4339. Do I need to add "in Lean 4"?

Comment (jcommelin) [reply Mathlib/FieldTheory/SplittingField/IsSplittingField.lean:59]: I just don't know what these lines mean. What breaks if you remove them? Maybe that helps with writing a more useful porting note?

Comment (Ruben-VandeVelde) [reply Mathlib/FieldTheory/SplittingField/IsSplittingField.lean:59]: The issue is that the f argument becomes implicit

Comment (jcommelin) [reply Mathlib/FieldTheory/SplittingField/IsSplittingField.lean:59]: Aha, got it.

Comment (jcommelin) [reply Mathlib/FieldTheory/SplittingField/IsSplittingField.lean:59]:
```suggestion
-- Porting note: infer kinds are unsupported
-- so we provide a version of `splits'` with `f` explicit.
```

File: Mathlib/FieldTheory/SplittingField/IsSplittingField.lean [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+namespace IsSplittingField
+
+variable {K}
+
+-- Porting note: infer kinds are unsupported
+theorem splits (f : K[X]) [IsSplittingField K L f] : Splits (algebraMap K L) f :=
+  splits'
+#align polynomial.is_splitting_field.splits Polynomial.IsSplittingField.splits
+
+-- Porting note: infer kinds are unsupported
```
Comment (jcommelin) [inline Mathlib/FieldTheory/SplittingField/IsSplittingField.lean:64]:
```suggestion
-- Porting note: infer kinds are unsupported
-- so we provide a version of `adjoin_roots'` with `f` explicit.
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**